obspy/core/inventory/util.py

```diff
@@ -1115,16 +1115,30 @@ def _add_resolve_seedid_doc(func):
     return func
 
 
+def _add_resolve_seedid_ph2comp_doc(func):
+    """
+    :param dict ph2comp: mapping of phases to components if format does not
+        specify the component or if the component ends with '?'. Set it to
+        `None` for no mapping of components. (default: {'P': 'Z', 'S': 'N'})
+    """
+    func.__doc__ = func.__doc__ + __doc__
```

Comment:
Check if `func.__doc__ is None`, or this will crash when run with optimization enabled.

Reply:
Thank you, I didn't know that doc strings were eliminated with optimization. Might also be applicable to other parts of the code base.